amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```diff
@@ -96,68 +103,57 @@ public TaskResult call() {
+        // We should not throw exception because it will trigger another ShutdownTask which has the same empty list of childShards.
+        // Instead we should shut down the ShardConsumer with LEASE_LOST reason to allow other worker to grab lease and continue reading from this shard.
+        if(reason == ShutdownReason.SHARD_END && CollectionUtils.isNullOrEmpty(childShards)) {
+            localReason = ShutdownReason.LEASE_LOST;
+            dropLease();
+            log.warn("False shard end encountered. Forcing the lease to be lost before shutting down the consumer for Shard: {}.", shardInfo.shardId());
+        } else {
+            localReason = reason;
+        }
+
         try {
             try {
-                ShutdownReason localReason = reason;
-                List<Shard> latestShards = null;
-                /*
-                 * Revalidate if the current shard is closed before shutting down the shard consumer with reason SHARD_END
-                 * If current shard is not closed, shut down the shard consumer with reason LEASE_LOST that allows
-                 * active workers to contend for the lease of this shard.
-                 */
-                if (localReason == ShutdownReason.SHARD_END) {
-                    latestShards = shardDetector.listShards();
-
-                    //If latestShards is empty, should also shutdown the ShardConsumer without checkpoint with SHARD_END
-                    if (CollectionUtils.isNullOrEmpty(latestShards) || !isShardInContextParentOfAny(latestShards)) {
-                        localReason = ShutdownReason.LEASE_LOST;
-                        dropLease();
-                        log.info("Forcing the lease to be lost before shutting down the consumer for Shard: " + shardInfo.shardId());
-                    }
-                }
+                log.debug("Invoking shutdown() for shard {}, concurrencyToken {}. Shutdown reason: {}",
+                        shardInfo.shardId(), shardInfo.concurrencyToken(), localReason);
 
-                // If we reached end of the shard, set sequence number to SHARD_END.
+                final long startTime = System.currentTimeMillis();
                 if (localReason == ShutdownReason.SHARD_END) {
+                    // Create new lease for the child shards if they don't exist.
+                    createLeasesForChildShardsIfNotExist();
+
                     recordProcessorCheckpointer
                             .sequenceNumberAtShardEnd(recordProcessorCheckpointer.largestPermittedCheckpointValue());
                     recordProcessorCheckpointer.largestPermittedCheckpointValue(ExtendedSequenceNumber.SHARD_END);
-                }
-
-                log.debug("Invoking shutdown() for shard {}, concurrencyToken {}. Shutdown reason: {}",
-                        shardInfo.shardId(), shardInfo.concurrencyToken(), localReason);
-                final ShutdownInput shutdownInput = ShutdownInput.builder().shutdownReason(localReason)
```

Comment:
Where the ShutdownInput was used before?

Reply:
It's never used. Should be safe to remove